synapse/handlers/message.py

```diff
@@ -888,11 +893,12 @@ def _send_dummy_events_to_fill_extremities(self):
         """Background task to send dummy events into rooms that have a large
         number of extremities
         """
-
+        self._expire_rooms_to_exclude_from_dummy_event_insertion()
         room_ids = yield self.store.get_rooms_with_many_extremities(
-            min_count=10, limit=5
+            min_count=10,
+            limit=5,
+            room_id_filter=list(self._rooms_to_exclude_from_dummy_event_insertion),
```

Comment:
I don't think there's any need to copy the keys here:
```suggestion
            room_id_filter=self._rooms_to_exclude_from_dummy_event_insertion.keys(),
```